### PR TITLE
CRUD on plugin templates to left panel - Hydrator++

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds the ability to do CRUD operations on plugin templates from left nav.

### CODE REVIEW

File: cdap-ui/app/features/hydrator-beta/templates/create/popovers/plugin-templates.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<plugin-templates></plugin-templates>
```
Comment (elfenheart) [inline cdap-ui/app/features/hydrator-beta/templates/create/popovers/plugin-templates.html]: need license headers

File: cdap-ui/app/features/hydrator-beta/templates/create/popovers/plugin-delete-confirmation.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<div class="modal-header">
```
Comment (elfenheart) [inline cdap-ui/app/features/hydrator-beta/templates/create/popovers/plugin-delete-confirmation.html]: license headers

File: cdap-ui/app/features/hydrator-beta/controllers/create/popovers/plugin-templates-delete-ctrl.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+angular.module(`${PKG.name}.feature.hydrator-beta`)
```
Comment (elfenheart) [inline cdap-ui/app/features/hydrator-beta/controllers/create/popovers/plugin-templates-delete-ctrl.js]: license headers

File: cdap-ui/app/features/hydrator-beta/controllers/create/popovers/plugin-templates-create-edit-ctrl.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+angular.module(`${PKG.name}.feature.hydrator-beta`)
```
Comment (elfenheart) [inline cdap-ui/app/features/hydrator-beta/controllers/create/popovers/plugin-templates-create-edit-ctrl.js]: license headers

File: cdap-ui/app/directives/plugin-templates/plugin-templates.less [hunk 5]
```diff
@@ -0,0 +1,11 @@
+plugin-templates {
```
Comment (elfenheart) [inline cdap-ui/app/directives/plugin-templates/plugin-templates.less]: license headers

File: cdap-ui/app/directives/plugin-templates/plugin-templates.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+angular.module(`${PKG.name}.commons`)
```
Comment (elfenheart) [inline cdap-ui/app/directives/plugin-templates/plugin-templates.js]: license headers

File: cdap-ui/app/directives/my-popover/my-popover.less [hunk 7]
```diff
@@ -0,0 +1,16 @@
+body.theme-cdap {
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-popover/my-popover.less]: license headers

File: cdap-ui/app/directives/plugin-templates/services/plugin-template-store.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    let dispatcher = PluginTemplateDispatcherBeta.getDispatcher();
+    dispatcher.register('onInit', this.initStore.bind(this));
+    dispatcher.register('onSaveSuccessfull', this.setIsSaveSuccessfull.bind(this));
+    dispatcher.register('onCloseCommand', this.setIsCloseCommand.bind(this));
+    dispatcher.register('onReset', this.setDefaults.bind(this));
+  }
+
+  setDefaults() {
+    console.log('Templates Store reset');
```
Comment (elfenheart) [inline cdap-ui/app/directives/plugin-templates/services/plugin-template-store.js]: remove?

File: cdap-ui/app/features/hydrator-beta/controllers/create/leftpanel-ctrl.js [hunk 9]
```diff
@@ -167,6 +182,85 @@ class LeftPanelControllerBeta {
...
+                return mySettings.set('pluginTemplates', res);
+              })
+              .then( () => {
+                myAlertOnValium.show({
+                  type: 'success',
+                  content: 'Successfully deleted template ' + node.pluginTemplate
+                });
+                PluginActionsFactoryBeta.fetchTemplates({namespace: $stateParams.namespace});
+              });
+            $scope.$close();
```
Comment (elfenheart) [inline cdap-ui/app/features/hydrator-beta/controllers/create/leftpanel-ctrl.js]: shouldn't this $close be within the callback?

**Comment by ajainarayanan** [comment]
@elfenheart Updated license headers - RAT check style build: http://builds.cask.co/browse/CDAP-DRC3493-3

File: cdap-ui/app/features/hydrator-beta/controllers/create/popovers/plugin-templates-delete-ctrl.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+          return mySettings.set('pluginTemplates', res);
+        })
+        .then( () => {
+          myAlertOnValium.show({
+            type: 'success',
+            content: 'Successfully deleted template ' + node.pluginTemplate
+          });
+          PluginActionsFactoryBeta.fetchTemplates({namespace: $stateParams.namespace});
+        });
+      $scope.$close();
```
Comment (elfenheart) [inline cdap-ui/app/features/hydrator-beta/controllers/create/popovers/plugin-templates-delete-ctrl.js]: should this $close() be inside the callback?

**Comment by elfenheart** [comment]
:+1: LGTM

**Comment by elfenheart** [comment]
:+1: LGTM

**Comment by ajainarayanan** [comment]
Rebased upon latest develop. Bamboo build passed for RAT style check (http://builds.cask.co/browse/CDAP-DRC3493-7) . Tested it in on a cluster. Hence merging.

**Pull request merged**